internal/backend/remote-state/http/client_test.go

```diff
@@ -251,3 +255,136 @@ func TestHttpClient_IsLockingEnabled(t *testing.T) {
 		})
 	}
 }
+
+// Tests the Lock method for the HTTP client.
+// Test to see correct lock info is returned
+func TestHttpClient_lock(t *testing.T) {
+	stateLockInfoA := statemgr.LockInfo{
+		ID:        "ada-lovelace-state-lock-id",
+		Who:       "AdaLovelace",
+		Operation: "TestTypePlan",
+	}
+
+	trimString := func(str string) string {
+		// Anonymous Helper function to remove new line, tab
+		// and space characters from a string
+		space := regexp.MustCompile(`\s+`)
+		return space.ReplaceAllString(str, " ")
+	}
+
+	testCases := []struct {
+		name           string
+		lockMethod     string
+		lockInfo       *statemgr.LockInfo
+		handler        http.HandlerFunc
+		validateResult func(lockID string, errorMessage error)
+	}{
```

Comment:
I would prefer a more clear separation between test logic and test data (inputs and outputs). I think the following structure covers test needs in a slightly simpler way (you would also need to change test run logic inside a loop):

```suggestion
	inputLockInfo := statemgr.LockInfo{
		ID:        "ada-lovelace-state-lock-id",
		Who:       "AdaLovelace",
		Operation: "TestTypePlan",
	}

	testCases := []struct {
		name				string
		lockResponseStatus	int
		lockResponseBody		[]byte
		expectedLockInfo		*statemgr.LockInfo // or just an ID
		expectedErrorMsg		string
	}{
	...
```

Reply:
I have updated the tests base your suggestion